### PR TITLE
Project Setup and Configuration

### PR DESCRIPTION
**Pull request by sanazehra2001**
- Virtual Environment Setup
- Django Installation
- Project and App Setup
- PostgreSQL Configuration
- Django REST framework configuration

### CODE REVIEW

**Comment by aneeq8597** [comment]
add pycache  dir and db file in .gitignore file

File: todo/settings.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+# Build paths inside the project like this: BASE_DIR / 'subdir'.
+BASE_DIR = Path(__file__).resolve().parent.parent
+
+
+# Quick-start development settings - unsuitable for production
+# See https://docs.djangoproject.com/en/5.0/howto/deployment/checklist/
+
+# SECURITY WARNING: keep the secret key used in production secret!
+SECRET_KEY = 'django-insecure-(+5gi6r%ebcjra_qjgy$r3rp%sha-ad)hr$6#ibutha1_dd2c='
```
Comment (aneeq8597) [inline todo/settings.py]: we can create .env file and get these configs from environment variables

**Comment by aneeq8597** [comment]
create requirements.txt file which will contain packages used for this django project

File: toDoApp/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
```
Comment (aneeq8597) [inline toDoApp/migrations/0001_initial.py]: this is django default user model . 
lets create a custom user model and use here

**Comment by sanazehra2001** [comment]
Implemented suggested changes

**Pull request merged**